### PR TITLE
Beam collision

### PR DESCRIPTION
**Pull request by Thorakii**
Beams now stop at the first thing they collide with.

Also resized the projectiles to scale with the new size of the cars and weapons

### CODE REVIEW

File: src/player.py [hunk 1]
```diff
@@ -27,8 +27,8 @@ def __init__(
         self.sprite.center_y = 256
         self.input = input
         self.sprite_lists = sprite_lists
-        self.primary_weapon_transform = (15, 10, 0)
-        self.secondary_weapon_transform = (15, -10, 0)
+        self.primary_weapon_transform = (16, 10, 0)
+        self.secondary_weapon_transform = (16, -10, 0)
```
Comment (Thorakii) [inline src/player.py:31]: Fixed bug where the beam hit the player's own car in the secondary weapon position

**Comment by Thorakii** [comment]
[Beam collision logic](https://trello.com/c/0oRklefS/111-beam-collision-logic)

**Review by cspotcode — CHANGES_REQUESTED** [review]
A few minor things, but code runs great and does what it says on the tin!

File: src/projectile.py [hunk 2]
```diff
@@ -100,27 +108,56 @@ def on_collision_with_player(
...
-        for player in players_touching_projectile:
-            player: LinkedSprite[Player]
+        player: LinkedSprite[Player] = self._shorten_beam(players_touching_projectile)
+        if player != None:
             player.owner.take_damage(self.damage * delta_time)
 
+    def _shorten_beam(self, collision_list: list[LinkedSprite]):
+        """
+        Given a list of sprites the beam collided with, stop the beam at that sprite
+        Return the first sprite that is collided with
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/projectile.py:141]: I like the suggestion. Updated

File: src/projectile.py [hunk 3]
```diff
@@ -100,27 +108,56 @@ def on_collision_with_player(
...
+        """
+        collisions = arcade.SpriteList()
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/projectile.py:143]: I agree, I hate doing this, but I have access to a list of sprites and an arcade function I need to run requires a spritelist. There has to be a better way to making a list of sprites into a spritelist, but I couldn't find it

File: src/projectile.py [hunk 4]
```diff
@@ -100,27 +108,56 @@ def on_collision_with_player(
...
+        closest_collision: arcade.Sprite = None
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/projectile.py:144]: Addeds optional, but slight change from the suggestion, the linked sprite is not always a Player. This code gets run on walls too.

File: src/projectile.py [hunk 5]
```diff
@@ -100,27 +108,56 @@ def on_collision_with_player(
...
+        for collision in collision_list:
+            collisions.append(collision)
+        point: Tuple(float, float) = self.muzzle_location[:2]
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/projectile.py:147]: Fixed

File: src/projectile.py [hunk 6]
```diff
@@ -100,27 +108,56 @@ def on_collision_with_player(
...
+        for x in range(1,self.beam_range):
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/projectile.py:148]: Ran formatter

File: src/projectile.py [hunk 7]
```diff
@@ -100,27 +108,56 @@ def on_collision_with_player(
...
+            if arcade.get_sprites_at_point(point, collisions):
+                closest_collision = arcade.get_sprites_at_point(point, collisions)[0]
+                self.sprite.width = x
+                break
+            point = add_vec(point, polar_to_cartesian(1, self.sprite.radians))
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/projectile.py:153]: Implimented

File: src/weapon.py [hunk 8]
```diff
@@ -165,9 +165,9 @@ class MachineGun(Weapon):
     weapon_icon = MACHINE_GUN
 
     def setup(self):
-        self.bullet_speed = 500
+        self.bullet_speed = 800
```
[the remark this answers is not in the document]
Comment (Thorakii) [reply src/weapon.py:168]: Yeah, we'll have to optimize all of the parameters like this for the most fun gameplay

**Review by cspotcode — APPROVED** [review]
Shipit! :shipit:

**Pull request merged**